docs/proposals/groups.md

```diff
@@ -0,0 +1,94 @@
+
+The Organization has a list members.  Only users from this list may be assigned to OrgGroups.  This prevents an overzelous organization admin from creating OrgGroups made up on non-members.  By default, an organization admin does not have rights to change Organization membership.
+
+*****TBD*****: How are users added to the Organization???  Being able to modify the OrgGroup structure of known members seem logically distinct from being able to force users into an Organization.  I wouldn't expect the same person to do both.
+
+Organizations will have an `organizations/groups` subresource.  A `GET` returns the list of OrgGroups.  A `POST` with a OrgGroup creates a new OrgGroup.  A `PUT` with a OrgGroup updates a new OrgGroup.  Subresources can't be named, so we do not have an individual get for `organizations/groups`.  This approach will require enhancements to `kubectl`, but prevents a Organization admins from being able to perform a `list` operation against `orggroups`.  The alternative is to have `list` on `orggroups` only show the `OrgGroups` you're allowed to see, but that would be different from all other `list`s.
+
+
+## Tying Projects to Organizations
+A Project may have at most one owner (Organization).  An Organization may own many projects.  We will indicate project ownership by having an "openshift.io/owner" label that contains a value of the form "resource/namespace/name".  Initially, the only valid resource will an Organization, so the value will look like: "organizations//MyOrg".  This allows for easy listing of projects owned by a particular Organization and allows us to extend the concept of project ownership to other resource types at a later time.
+
+
+## Using OrgGroups in RoleBindings
+RoleBindings will get a new field:
+```
+	// OrgGroups holds all the OrgGroups directly bound to the role.
+	OrgGroups util.StringSet
+```
+The `OrgGroups` field can only refer to OrgGroups owned by the Organization that owns the Project.  Since OrgGroup names are always scope qualified, any disallowed OrgGroup name will be rejected by the RESTStorage and will not be respected by our authorizer.  Our `oc policy` commands will be able to automatically scope qualify OrgGroups.
+
+
+## UserInfo and OrgGroups
+UserInfo is an interface shared with kubernetes.  If they won't change their interface, we'll be able to re-use it in a backwards compatible way by scope qualifying our OrgGroup names as ClusterGroups.  This means that being a member of OrgGroup "MyOrg:AOrgGroup" would result in a ClusterGroup membership that looks like "org:MyOrg:AOrgGroup".  The only signficant side effect is that RoleBinding RESTStorage must reject ClusterGroups that start with "org:", but those ClusterGroups are invalid ClusterGroup names anyway.
```

Comment:
Isn't the UserInfo actually `user.Info` from `GoogleCloudPlatform/kubernetes/pkg/auth/user`.
That's again covering my previous question, why not always using ClusterGroups with the `org` prefix?